content/src/main/content/jcr_root/apps/acs-commons/touchui-widgets/configure-limit-parsys/touchui-limit-parsys.js

```diff
@@ -82,27 +94,19 @@
         return children;
     }
 
-    function isWithinLimit(editable){
-        var path = getDesignPath(editable),
-            children = getChildEditables(editable.getParent()),
+    function isWithinLimit(parsysEditable){
+        var children = getChildEditables(parsysEditable),
             isWithin = true, currentLimit = "";
 
-        if (path !== undefined) {
-            $.ajax({url: path + ".2.json", async: false}).done(function (data) {
-                if (_.isEmpty(data) || !data[ACS_COMPONENTS_LIMIT]) {
-                    return;
-                }
-
-                currentLimit = data[ACS_COMPONENTS_LIMIT];
-
-                var limit = parseInt(currentLimit);
-
-                isWithin = children.length <= limit;
-            });
+        currentLimit = _findPropertyFromDesign(parsysEditable, Granite.author.pageDesign, ACS_COMPONENTS_LIMIT);
+        if (currentLimit === null) {
+            return false;
```

Comment:
We might need to return true  in this line instead of false. If no current limit is set for the parsys the insert dialog is not shown.

Reply:
Good hint, fixed!